test/convection/model/test_validation.rb

```diff
@@ -0,0 +1,184 @@
+      description 'Validations Test Template - Excessive Resource Name'
+        logs_log_group "0"*256  do
+          retention_in_days 365
+        end
+      end
+    @excessive_mappings = ::Convection.template do
+      description 'Validations Test Template - Too Many Mappings'
+      101.times do |count|
+        mapping "Mapping_#{count}" do
+          item 'us-east-1', 'test', 'cf-test-keys'
+          item 'us-west-1', 'test', 'cf-test-keys'
+        end
+      end
+    end
+    @excessive_mapping_attributes = ::Convection.template do
+      description 'Validations Test Template - Too Many Mapping Attributes'
+      mapping "Mapping_Example" do
+        31.times do |count|
+          item "#{count}_1", "#{count}_2", "#{count}_3"
+        end
+      end
+    end
+    @excessive_mapping_name = ::Convection.template do
+      description 'Validations Test Template - Excessive Mapping Name'
+      mapping "0" *256 do
+        item "1","2","3"
+      end
+    end
+    @excessive_mapping_attribute_names = ::Convection.template do
+      description 'Validations Test Template - Excessive Mapping Attribute Name'
+      mapping "Mapping_1" do
+        item "0" *256,"value","value"
+      end
+    end
+    @excessive_outputs = ::Convection.template do
+      description 'Validations Test Template - Too Many Outputs'
+      61.times do |count|
+        output "Output_#{count}" do
+          description 'An Important Attribute'
+          value get_att('Resource', 'Attribute')
+        end
+      end
+    end
+    @excessive_parameters = ::Convection.template do
+      description 'Validations Test Template - Too Many Parameters'
+      61.times do |count|
+        parameter "Parameter_#{count}" do
+          type 'String'
+          description 'Example Parameter'
+          default 'm3.medium'
+        end
+      end
+    end
+
+    @excessive_parameter_name = ::Convection.template do
+      description 'Validations Test Template - Excessive Parameter Name'
+      parameter "0" * 256 do
+        type 'String'
+        description 'Example Parameter'
+        default 'm3.medium'
+      end
+    end
+    @excessive_parameter_value_bytesize  = ::Convection.template do
+      description 'Validations Test Template - Excessive Parameter Value Bytesize'
+      parameter "Excessive_Parameter" do
+        type 'String'*150
+        description 'Example Parameter'*150
+        default 'm3.medium'*150
+      end
+    end
+    @excessive_output_name = ::Convection.template do
+      description 'Validations Test Template - Excessive Output Name'
+      output "0" * 256 do
+        description 'An Important Attribute'
+        value get_att('Resource', 'Attribute')
+      end
+    end
+  end
+
+  def test_within_limits
+    assert_nothing_raised(ArgumentError) {@within_limits.to_json}
+  end
+  def test_bytesize
+    assert_raises(ArgumentError, "Error: Excessive Template Size") {JSON.generate(@excessive_bytesize.to_json).bytesize}
```

Comment:
Spaces after `{` and before `}` for single line blocks. Note some of these blocks are pretty long. Consider using the `do`/`end` format instead.